model/map/src/main/java/org/keycloak/models/map/deploymentState/MapDeploymentStateProviderFactory.java

```diff
@@ -50,13 +50,18 @@ public void init(Config.Scope config) {
         String seed = config.get(RESOURCES_VERSION_SEED);
         if (seed == null) {
             Logger.getLogger(DeploymentStateProviderFactory.class)
-                    .warnf("It is recommended to set '%s' property in the %s provider config of %s SPI", RESOURCES_VERSION_SEED, PROVIDER_ID, DeploymentStateSpi.NAME);
+                    .warnf("Version seed for deployment state set with a random number. Caution: This can lead to unstable operations when serving resources from the cluster without a sticky loadbalancer or when restarting nodes. Set the '%s' property in the %s provider config of %s SPI for stable operations", RESOURCES_VERSION_SEED, PROVIDER_ID, DeploymentStateSpi.NAME);
             //generate random string for this installation
             seed = SecretGenerator.getInstance().randomString(10);
         }
         try {
+            StringBuilder toDigest = new StringBuilder(seed);
+            toDigest.append("-").append(new ModelVersion(Version.VERSION_KEYCLOAK));
+            if (Version.VERSION_KEYCLOAK.endsWith("-SNAPSHOT")) {
+                toDigest.append("-").append(Version.BUILD_TIME);
+            }
```

Comment:
What benefit has this method as compared to the previous one?

Reply:
The previous version ignored the SNAPSHOT at the end of `VERSION_KEYCLOAK`. It mimics the behavior in `Version`, which will add the build time. 

Using this approach, a new `RESOURCES_VERSION` is created for each snapshot aka nightly build.

Reply:
This only adds extra salt to an already randomly generated seed. Hence any two runs of the snapshot would already have a different RESOURCES_VERSION already, and no snapshot version should be run in production where it would matter. Thus I struggle to see any extra benefit. Please revert.

Reply:
The situation you describe will only apply when a SNAPSHOT version is run without the parameter `--storage-deployment-state-version-seed`. 

When running load tests or exploratory tests with a nightly build for example as part of keycloak-benchmark, I'd expect the parameter to be set. Once this parameter is being set, the `RESOURCES_VERSION` should change with newly deployed nightly build, therefore I added the `BUILD_TIME` as done in the Version - see below for the code in Version.java.

https://github.com/keycloak/keycloak/blob/main/common/src/main/java/org/keycloak/common/Version.java#L52-L54

Reply:
Ah, now I can see where you are coming from. So to prevent copy-pasting th logic, the original code should keep the calculation of the `toDigest` based on (then unmangled `Version.RESOURCES_VERSION`) rather than `Version.VERSION_KEYCLOAK`, and that would do the same job, correct?

Sidenote: any cluster (even the same nightly build) must be deployed with a fresh seed value, hence two consecutive snapshot versions would always have a different `Version.RESOURCES_VERSION`.

Reply:
Yes, I will try to use the `Version.RESOURCES_VERSION` for this; I still need to double-check what is preserved by the `ModelVersion` class that parses the value. Depending on the result I might use the `Version.RESOURCES_VERSION` as is.

> Sidenote: any cluster (even the same nightly build) must be deployed with a fresh seed value, hence two consecutive snapshot versions would always have a different Version.RESOURCES_VERSION.

Could you please elaborate on the scenario when this comes into play? I haven't yet seen a code snippet that I can connect to that note, and I fear that the warning and description might not reflect the situation when the seed should be changed.

Reply:
pushed changes to `MapDeploymentStateProviderFactory` - as `ModelVersion` wasn't able to parse `RESOURCES_VERSION` with the dash, I now take the `RESOURCES_VERSION` as is together with the seed. 

The question in the previous comment still applies; still the answer might be handled in a different PR.

Reply:
> pushed changes to `MapDeploymentStateProviderFactory` - as `ModelVersion` wasn't able to parse `RESOURCES_VERSION` with the dash, I now take the `RESOURCES_VERSION` as is together with the seed.

Thank you!

> > Sidenote: any cluster (even the same nightly build) must be deployed with a fresh seed value, hence two consecutive snapshot versions would always have a different Version.RESOURCES_VERSION.
> 
> Could you please elaborate on the scenario when this comes into play? I haven't yet seen a code snippet that I can connect to that note, and I fear that the warning and description might not reflect the situation when the seed should be changed.

This a reaction re the sample scenario:

> When running load tests or exploratory tests with a nightly build for example as part of keycloak-benchmark, I'd expect the parameter to be set. Once this parameter is being set, the RESOURCES_VERSION should change with newly deployed nightly build, therefore I added the BUILD_TIME as done in the Version - see below for the code in Version.java.

The seed might be kept for the same deployment. Perhaps I should have said "every distinguishable cluster" - if there are multiple configurations of the same nightly build cluster run consecutively, it seems about correct to treat them as distinguishable, and thus also set a fresh seed. So specifically for keycloak-benchmark, I'd go with a fresh seed value (consistent across the cluster) everytime the benchmark is run.

However for the same deployment (say a production cluster) which keeps about the same configuration, the seed can remain unchanged in time.